docs/enhancements/soft-install-timeout.md

```diff
@@ -0,0 +1,176 @@
+---
+title: soft-install-timeout
+authors:
+- "@jhernand"
+- "@oamizur"
+creation-date: 2023-10-04
+last-updated: 2023-11-23
+---
+
+# Soft install timeout
+
+## Summary
+
+Currently cluster installation by assisted installer may fail due to timeout expiration.
+Assisted installer maintains many timeouts with different values.  A timeout limits the time
+period to run an installation stage or to perform a specific installation operation. 
+We want to change these timeouts to be soft.  Soft timeout expiration will cause warning that the
+installation is taking longer than expected and the installation will continue.
+
+## Motivation
+
+This is important because users may be able to fix the issues that delay the
+installation instead of having to start it over.  This is not true for ZTP installations.  Users using ZTP
+will not try to fix these installations in case of a failure.
+
+### Goals
+
+### Non-Goals
+
+It is not a goal to change the global installation timeout configured via the
+`INSTALLATION_TIMEOUT` environment variable. That is set by default to `24h`
+and at that point the assisted service stops monitoring the cluster.
+
+## Proposal
+
+### User Stories
+
+#### Allow installation in slow network environments
+
+A user having installations in slow network environment that may take long enough that they can triggered the timeout. 
+I want my installations to continue and succeed even if they are very slow.
+
+#### Allow manually fixing a SaaS installation
+
+As a user I tried to create a SNO cluster in the SaaS environment. An issue
+prevented one of the cluster operators from reporting success. After one hour
+the timeout expired and the installation was marked as failed. When I found the
+failed install, I was able to easily resolve the issue and get the cluster
+operator to report success. But the service no longer cared; as far as it was
+concerned, the installation failed and was permanently marked as such. It also
+would not give me the kubeadmin password, which is an important feature of the
+install experience and tricky to obtain otherwise. I would like the service
+inform me about the exceeded timeout, but it should give me the kubeadmin
+password (if available at that point) and after my fixes it should continue
+and eventually mark the installation as successful.
+
+#### Allow manually fixing a late binding installation
```

Comment:
I'm not sure what late binding has to do with this feature.

Are we explicitly disabling the soft-timeouts for "early" binding even if the feature is enabled?

Reply:
I agree.  Will remove the paragraph